cmd/cortex/main.go

```diff
@@ -106,6 +110,11 @@ func main() {
 		}
 	}
 
+	if printVersion {
+		fmt.Fprintln(os.Stdout, version.Print("Cortex"))
```

Comment:
Cool, but how is it actually replaced with the real version?

Reply:
Our build process [injects](https://github.com/cortexproject/cortex/blob/82ca21840949a3137be22f7a5e48426751615d98/Makefile#L92) version, revision and branch to [variables in main.go](https://github.com/cortexproject/cortex/blob/195bac9900f8456003ef16560328c0719f342d8c/cmd/cortex/main.go#L27), which are then pushed to [`version` package in `init` method](https://github.com/cortexproject/cortex/blob/195bac9900f8456003ef16560328c0719f342d8c/cmd/cortex/main.go#L43). By the time we print version here, values are correct.

Reply:
Nice!